### DISCUSSION TITLE
Check that a header is well passed to an http mock

### DISCUSSION THREAD

**Discussion by ibrahimhammanid** [Decathlon/tzatziki#134] [Q&A]
Hello All
I have a testing scenario that I can't get to work; I seek your help please:

- First I have call to do, to my main micro-service
- That main micro-service needs to call a side micro-service, and passes to it a custom header, let's call it : My-Awesome-Header
- Now I mock my side microservice requests and response
- and I check that the mock has been called

till now everything works fine; but I want to check one last thing:

- I need to ensure that my main micro-service sent My-Awesome-Header while calling the side micro-service.

I tried a lot of solutions as explained in the tzatziki-http  module,  but nothing seems to work for me.
please find below a sample of my gherkin I modified on purpose for the sake of simplicity:
```Gherkin
Feature: Check my header

  Background:
    * that "my/awesome/side/microservice" is mocked as:
  """yaml
    request:
      method: GET
    response:
      status: OK_200
      body:
        payload:
          message: "What an awesomeness"
    """

  Scenario: Use my main microservice that calls my side microservice
    When postman_MS sends on "/my/awesome/main/microservice":
    """yaml
    method: POST
    body:
      payload:
        message: "how do I did?"

    """
    Then the interaction on "my/awesome/side/microservice" was:
    """yaml
      request:
        headers:
          My-Awesome-Header: {{_request.headers.entries.0.values.0.}}
    """
```
If you need more informations please feel free to ask me

**Comment by brian-mulier** [comment]
Hello Ibrahim,

`{{_request.headers.entries.0.values.0.}}` this part seems odd to me, what are you trying to achieve exactly ? The way you did it seems perfectly fine, however I can't get which value you expect to have there, are you trying to ensure the headers you gave as input to your main microservice is passed-through to your side one ?

**Reply by ibrahimhammanid** [reply]
Hi thanks for your reply,
I want to ensure that my main micro-service passes a custom header to my side micro-service; the issue is that {{_request}} is never evaluated , and it's used as a string in my gherkin
